test/functional/lua/loop_spec.lua

```diff
@@ -159,4 +159,34 @@ describe('vim.uv', function()
   it("is equal to require('luv')", function()
     eq(true, exec_lua("return vim.uv == require('luv')"))
   end)
+
+  it('non-string error() #32595', function()
+    local screen = Screen.new(50, 10)
+    exec_lua(function()
+      local timer = assert(vim.uv.new_timer())
+      timer:start(0, 0, function()
+        timer:close()
+        error()
+      end)
+    end)
+    local s = [[
+                                                        |
+      {1:~                                                 }|*5
+      {3:                                                  }|
+      {9:Error executing callback:}                         |
+      {9:[NULL]}                                            |
+      {6:Press ENTER or type command to continue}^           |
+    ]]
+    screen:expect(s)
+    feed('<cr>')
+    n.assert_alive()
+    exec_lua(function()
+      vim.uv.fs_stat('foo', function()
```

Comment:
Should we use a name that is less likely to have a false positive (from some other test or whatever)?
```suggestion
      vim.uv.fs_stat('non-existent-file', function()
```

Reply:
Don't matter? `callback` will run `error()` no matter file exist or not.